07-routing-gossip.md

```diff
@@ -0,0 +1,252 @@
+`node-id-2` using `bitcoin-key-2`
+
+The creating node MUST set `node-signature-1` to the signature of the
+double-SHA of every data field up to and including
+`bitcoin-signature-2` using `node-id-1`, and `node-signature-2` to the
+signature of the double-SHA of every data field up to and including
+`bitcoin-signature-2` using `node-id-2`.
+
+The receiving node MUST ignore the message if the output `outputindex`
+in block `blockheight` transaction number `blockindex` does not
+correspond to a P2WSH using `bitcoin-key-1` and `bitcoin-key-2` as
+specified in [BOLT #3](03-transactions.md#funding-transaction-output).
+The receiving node MUST ignore the message if this output is spent.
+
+Otherwise, the receiving node SHOULD fail the connection if
+`bitcoin-signature-1`, `bitcoin-signature-2`, `node-signature-1` or
+`node-signature-2` are invalid or not correct.
+
+Otherwise, if `node-id-1` or `node-id-2` are blacklisted, it SHOULD
+ignore the message.
+
+Otherwise, if the transaction referred to was not previously announced
+as a channel, the receiving node SHOULD queue the message for
+rebroadcasting.  If it has previously received a valid
+`channel_announcement` for the same transaction in the same block, but
+different `node-id-1` or `node-id-2`, it SHOULD blacklist the the
+previous message's `node-id-1` and `node-id-2` as well as this
+`node-id-1` and `node-id-2` and forget channels conntected to them,
+otherwise it SHOULD store this `channel_announcement`.
+
+The receiving node SHOULD forget a channel once its funding output has
+been spent or reorganized out.
+
+## Rationale
+
+Requiring both nodes to sign indicates they are both willing to route
+other payments via this node (ie. take part of the public network).
+Requiring the bitcoin signatures proves they control the channel.
+
+The blacklisting of conflicting nodes means that we disallow multiple
+different announcements: no node should ever do this, as it implies
+that keys have leaked.
+
+While channels shouldn't be advertised before they are sufficiently
+deep, the requirement against rebroadcasting only applies if the
+transaction hasn't moved to a different block.
+
+## The `node_announcement` message
+
+This allows a node to indicate extra data associated with it, in
+addition to its public key.  To avoid trivial denial of service attacks,
+nodes for which a channel is not already known are ignored.
+
+1. type: 257 (`MSG_NODE_ANNOUNCEMENT`)
+2. data:
+   * [64:signature]
+   * [4:timestamp]
+   * [16:ipv6]
+   * [2:port]
+   * [33:node-id]
+   * [3:rgb-color]
+   * [2:pad]
+   * [4:len]
+   * [len:alias]
+
+The `timestamp` allows ordering in the case of multiple announcements;
+the `ipv6` and `port` allow the node to announce its willingness to
+accept incoming network connections, the `rgb-color` and `alias` allow
+intelligence services to give their nodes cool monikers like IRATEMONK
+and WISTFULTOLL and use the color black.
+
+### Requirements
```

Comment:
Should a `node_announcement` message be sent after every new `channel_announcement`, or once after the first `channel_announcement`?